cloud-init/write-kali-aws-config.tpl.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+# This is a Terraform template file, and the input variables are
+# passed in via templatefile().
+#
+# shellcheck disable=SC2154
+
+# This script is used to write out an AWS configuration to the home
+# directory of a specified user.  The AWS configuration gives the user
+# access to the role that allows writing to the findings data S3
+# bucket.
+
+# Input variables are:
+# * aws_region - the AWS region where the roles are to be assumed
+# * findings_data_bucket_write_role_arn - the ARN of the IAM role that
+#   can be assumed to write to the findings data S3 bucket
+# * permissions - the octal permissions to assign the AWS
+#   configuration
+# * vnc_read_parameter_store_role_arn - the ARN of the role that
+#   grants read-only access to certain VNC-related SSM Parameter Store
+#   parameters, including the VNC username
+# * vnc_username_parameter_name - the name of the SSM Parameter Store
+#   parameter containing the VNC user's username
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Temporarily assume a role in order to retrieve the VNC user's
+# username from SSM Parameter Store.
+aws_sts_output=$(aws sts assume-role \
+  --role-arn="${vnc_read_parameter_store_role_arn}" \
+  --role-session-name=cloud-init)
+access_key_id=$(sed --quiet \
+  's/^[[:blank:]]*"AccessKeyId": "\([[:graph:]]\+\)",$/\1/p' \
+  <<< "$aws_sts_output")
+secret_access_key=$(sed --quiet \
+  's/^[[:blank:]]*"SecretAccessKey": "\([[:graph:]]\+\)",$/\1/p' \
+  <<< "$aws_sts_output")
+session_token=$(sed --quiet \
+  's/^[[:blank:]]*"SessionToken": "\([[:graph:]]\+\)",$/\1/p' \
+  <<< "$aws_sts_output")
+
+# Now retrieve the VNC user's username from SSM Parameter Store
+ssm_response=$(AWS_ACCESS_KEY_ID=$access_key_id \
+  AWS_SECRET_ACCESS_KEY=$secret_access_key \
+  AWS_SESSION_TOKEN=$session_token \
+  aws --region "${aws_region}" \
+  ssm get-parameter \
+  --name /vnc/username --with-decryption)
+vnc_username=$(sed --quiet \
+  's/^[[:blank:]]*"Value": "\([[:graph:]]\+\)",$/\1/p' \
+  <<< "$ssm_response")
```

Comment:
If all we need is the VNC username can't we just take it as a template argument and use the value of https://github.com/cisagov/cool-assessment-terraform/blob/513726d23126a20b11c23dfb70439e1a6d90755e/locals.tf#L44-L48 like we do in https://github.com/cisagov/cool-assessment-terraform/blob/513726d23126a20b11c23dfb70439e1a6d90755e/terraformer_cloud_init.tf#L44? This also moves away from hard-coding the parameter name.

```suggestion
```